### PR TITLE
Split SYSTEMD_ENABLED to PRIVILEGED_ENABLED and CGROUP_MOUNT_ENABLED

### PR DESCRIPTION
**Pull request by gaol**
This relates to issue: https://issues.redhat.com/browse/SET-422, but if anyone think this deserves a new issue, I can create one for it.

### CODE REVIEW

**Comment by gaol** [comment]
BTW, from https://docs.podman.io/en/latest/markdown/podman-run.1.html#systemd-true-false-always, the `--systemd=true` is the default value.

**Comment by gaol** [comment]
@rpelisse would you please review ? thanks

File: run.sh [hunk 1]
```diff
@@ -37,9 +38,15 @@ add_ports_if_provided() {
  fi
 }
 
-systemd_if_enabled() {
-  if [ -n "${SYSTEMD_ENABLED}" ]; then
-    echo "--systemd=true --privileged=true -v /sys/fs/cgroup:/sys/fs/cgroup:ro"
+privileged_if_enabled() {
+  if [ -n "${PRIVILEGED_ENABLED}" ]; then
+    echo "--systemd=true --privileged=true"
```
Comment (rpelisse) [inline run.sh:43]: @gaol let's be complete about our refactoring! Can you add a system_if_enabled() function? So we'll have one function for each (cgroup, sysd, priv).

File: run.sh [hunk 2]
```diff
@@ -83,7 +90,7 @@ readonly CONTAINER_COMMAND=${CONTAINER_COMMAND:-"${WORKSPACE}/hera/wait.sh"}
 run_ssh "podman run \
             --name "${CONTAINER_TO_RUN_NAME}" $(container_user_if_enabled) \
             --add-host=${CONTAINER_SERVER_HOSTNAME}:${CONTAINER_SERVER_IP}  \
-            --rm $(add_parent_volume_if_provided) $(systemd_if_enabled) \
+            --rm $(add_parent_volume_if_provided) $(privileged_if_enabled) $(cgroup_mount_if_enabled) \
```
Comment (rpelisse) [inline run.sh:93]: I'll move this to a newline.

**Review by rpelisse — CHANGES_REQUESTED** [review]
@gaol a few more enhancement, but otherwise it's look good.

File: run.sh (hunk 2 above)
Comment (gaol) [reply run.sh:93]: Moving to a newline will lead to double `\` in case they are **not** defined.

**Review by rpelisse — APPROVED** [review]
@spyrkob LGTM! any comment?

**Comment by spyrkob** [comment]
LGTM

**Pull request merged**